### PR TITLE
REST Endpoint Phase 2 - Corrected

### PR DESCRIPTION
**Pull request by logotie**
On submitting a post request to /api/torrentfile with the JSON ->
{ "url":""}
'url' field containing a valid url that points to a torrent file.

You will receive back
{
"success":"true",
"id":"#####",
"error":""
}

If the url is not valid, you will back an error: 'Web url is not valid'.
If the file at the url, is not a torrent file you will receive back: 'File at url address is not a torrent file'

### CODE REVIEW

**Comment by Earu** [comment]
Required changes:
- https://github.com/logotie/Sulfur/blob/master/Sulfur/Constants/ServiceConstants.cs
In the above file, enums in C# use `CamelCase` as naming convention, 
For the static variables you use for your exception, add them to your resources file.

- https://github.com/logotie/Sulfur/blob/master/Sulfur/Services/UrlFileDownloadActions/UrlFileDownloadService.cs
You can create tuples like so: 
```cs
(int component1, string component2)
```
Instead of using Tuple.Create, you can also make a class which is more idiomatic of C# lang

**Comment by logotie** [comment]
Alright, I made the changes

**Comment by Earu** [comment]
You need to use the tuple construct like so:
```cs
private (int, string) ReturnATuple()
{
    return (1, "cool");
}
```

The `CreateAndReturn` util method is needed here.

**Comment by logotie** [comment]
Wow, didn't know one could do that haha

Pretty awesome!

File: Sulfur/Services/UrlFileDownloadActions/UrlFileDownloadService.cs [hunk 1]
```diff
@@ -32,9 +32,9 @@ public class UrlFileDownloadService : IUrlFileDownloadService
         }
 
         //Util method to create a tuple.
-        private Tuple<bool, ServiceConstants.UrlFileDlEnums> CreateAndReturnTuple(bool endResult, ServiceConstants.UrlFileDlEnums resultEnum)
+        private (bool, ServiceConstants.UrlFileDlEnums) CreateAndReturnTuple(bool endResult, ServiceConstants.UrlFileDlEnums resultEnum)
```
Comment (Earu) [inline Sulfur/Services/UrlFileDownloadActions/UrlFileDownloadService.cs]: Again, you dont need this method, it only makes your code more verbose when it does not need to be

Comment (logotie) [reply Sulfur/Services/UrlFileDownloadActions/UrlFileDownloadService.cs]: Ah apologies I misunderstood the first time, I think I have fixed it now

**Pull request merged**